types/node/index.d.ts

```diff
@@ -772,23 +773,29 @@ declare module "http" {
     import * as stream from "stream";
     import { URL } from "url";
 
-    export interface RequestOptions {
+    export interface ClientRequestArgs {
         protocol?: string;
         host?: string;
         hostname?: string;
         family?: number;
         port?: number | string;
+        defaultPort?: number | string;
         localAddress?: string;
         socketPath?: string;
         method?: string;
         path?: string;
         headers?: { [key: string]: any };
         auth?: string;
         agent?: Agent | boolean;
+        _defaultAgent?: Agent,
         timeout?: number;
+        rejectUnauthorized?: boolean;
```

Comment:
Yes, because ClientRequest class is reused by HTTPS. Setting this flag for plain http doesn't make much sense as it is related to tls.